custom_components/detailed_hello_world_push/cover.py

```diff
@@ -0,0 +1,160 @@
+    # identical "identifiers" attribute, but not return a name attribute.
+    # See the sensors.py file for the corresponding example setup.
+    # Additional meta data can also be returned here, including sw_version (displayed
+    # as Firmware), model and manufacturer (displayed as <model> by <manufacturer>)
+    # shown on the device info screen. The Manufacturer and model also have their
+    # respective columns on the Devices overview table. Note: Many of these must be
+    # set when the device is first added, and they are not always automatically
+    # refreshed by HA from it's internal cache.
+    # For more information see:
+    # https://developers.home-assistant.io/docs/device_registry_index/#device-properties
+    @property
+    def device_info(self):
+        """Information about this entity/device."""
+        return {
+            "identifiers": {(DOMAIN, self._roller.roller_id)},
+            # If desired, the name for the device could be different to the entity
+            "name": self.name,
+            "sw_version": "v0.0.1",
+            "model": "Test Device",
+            "manufacturer": "Demonstration Corp",
+        }
+
+    # This is the name for this *entity*, the "name" attribute from "device_info"
+    # is used as the device name for device screens in the UI. This name is used on
+    # entity screens, and used to build the Entity ID that's used is automations etc.
+    @property
+    def name(self):
+        """Return the name of the roller."""
+        return self._roller.name
+
+    # This property is important to let HA know if this entity is online or not.
+    # If an entity is offline (return False), the UI will refelect this.
+    @property
+    def available(self) -> bool:
+        """Return True if roller and hub is available."""
+        return self._roller.online and self._roller.hub.online
+
+    # The follwing properties are how HA knows the current state of the device.
+    # These must return a value from memory, not make a live query to the device/hub
+    # etc when called (hence they are properties). For a push based integration,
+    # HA is notified of changes via the async_write_ha_state call. See the __init__
+    # method for hos this is implemented in this example.
+    # The properties that are expected for a cover are based on the supported_features
+    # property of the object. In the case of a cover, see the following for more
+    # details: https://developers.home-assistant.io/docs/core/entity/cover/
+    @property
+    def current_cover_position(self):
+        """Return the current position of the cover."""
+        return self._roller.position
+
+    @property
+    def is_closed(self):
+        """Close the cover, same as position 0."""
+        return self._roller.position == 0
+
+    @property
+    def is_open(self):
+        """Open the cover, same as position 100."""
+        return self._roller.position == 100
+
```

Comment:
This is not an official one.
```suggestion
```

Reply:
Good point! :) 
I have updated and added a `is_opening` and `is_closing`.
Thanks again!